Adds Rock Entity Mapper

redreceipt opened the pull request:
## DESCRIPTION

### What does this PR do, or why is it needed?

This allows us to map other GraphQL types to Rock types besides content items.

### What design trade-offs/decisions were made?

### How do I test this PR?

### What automated tests did you write?

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

redreceipt commented:
tests are failing with this

![Screen Shot 2020-01-09 at 1 26 14 PM](https://user-images.githubusercontent.com/2659478/72094108-ab3ba800-32e3-11ea-9576-324359784414.png)

vinnyjth commented:
@redreceipt I updated your PR so it will fix tests, and also moved the function into the datasource so it can be overridden better in the future. Can you write a test for the new functionality (confirming mapping works if it exists in the config)?  Then I'll merge this

redreceipt commented:
> @redreceipt I updated your PR so it will fix tests, and also moved the function into the datasource so it can be overridden better in the future. Can you write a test for the new functionality (confirming mapping works if it exists in the config)? Then I'll merge this

done
